examples/core/quad/main.rs

```diff
@@ -288,9 +288,13 @@ fn main() {
     };
 
     // TODO: check transitions: read/write mapping and vertex buffer read
-    device.write_mapping::<Vertex>(&vertex_buffer, 0..buffer_len)
-        .unwrap()
-        .copy_from_slice(&QUAD);
+    {
+        let mut vertices = device
+            .acquire_mapping_writer::<Vertex>(&vertex_buffer, 0..buffer_len)
```

Comment:
I think we can provide better ergonomics here (on top of the current API), something like
```rust
device.map_scope(&vertex_buffer, 0..buffer_len)
    .unwrap()
    .copy_from_slice(&QUAD);
```
So `map_scope` would return a temporary object holding a mutable reference to the `device` and releasing the mapping automatically

I suspect we can get the type from the buffer format, since it's typed?

Reply:
The scope could be nice indeed, I was more thinking about a closure but I think that error handling might be less convenient with a closure.
This is the `core` example so there is no typed handle here.